contracts/access/manager/AccessManager.sol

```diff
@@ -568,44 +575,57 @@ contract AccessManager is Context, Multicall, IAccessManager {
      * operation is not yet scheduled, has expired, was executed, or was canceled.
      */
     function getSchedule(bytes32 id) public view virtual returns (uint48) {
-        uint48 timepoint = _schedules[id];
+        uint48 timepoint = _schedules[id].timepoint;
         return _isExpired(timepoint) ? 0 : timepoint;
     }
 
+    /**
+     * @dev Return the nonce for the latest scheduled operation with a given id. Returns 0 if the operation has never
+     * been scheduled.
+     */
+    function getNonce(bytes32 id) public view virtual returns (uint32) {
+        return _schedules[id].nonce;
+    }
+
     /**
      * @dev Schedule a delayed operation for future execution, and return the operation identifier. It is possible to
      * choose the timestamp at which the operation becomes executable as long as it satisfies the execution delays
      * required for the caller. The special value zero will automatically set the earliest possible time.
      *
      * Emits a {OperationScheduled} event.
      */
-    function schedule(address target, bytes calldata data, uint48 when) public virtual returns (bytes32) {
+    function schedule(address target, bytes calldata data, uint48 when) public virtual returns (bytes32 operationId, uint32 nonce) {
         address caller = _msgSender();
 
         // Fetch restriction to that apply to the caller on the targeted function
         (bool allowed, uint32 setback) = _canCallExtended(caller, target, data);
 
         uint48 minWhen = Time.timestamp() + setback;
 
+        if (when == 0) {
+            when = minWhen;
+        }
+
         // If caller is not authorised, revert
-        if (!allowed && (setback == 0 || when.isSetAndPast(minWhen - 1))) {
+        if (!allowed && (setback == 0 || when < minWhen)) {
             revert AccessManagerUnauthorizedCall(caller, target, bytes4(data[0:4]));
         }
 
         // If caller is authorised, schedule operation
-        bytes32 operationId = _hashOperation(caller, target, data);
+        operationId = _hashOperation(caller, target, data);
 
         // Cannot reschedule unless the operation has expired
-        uint48 prevTimepoint = _schedules[operationId];
+        uint48 prevTimepoint = _schedules[operationId].timepoint;
         if (prevTimepoint != 0 && !_isExpired(prevTimepoint)) {
             revert AccessManagerAlreadyScheduled(operationId);
         }
 
-        uint48 timepoint = when == 0 ? minWhen : when;
-        _schedules[operationId] = timepoint;
-        emit OperationScheduled(operationId, timepoint, caller, target, data);
+        nonce = _schedules[operationId].nonce + 1;
```

Comment:
What do you think?

```suggestion
        unchecked {
          nonce = _schedules[operationId].nonce + 1;
        }
```

Reply:
I'm not sure. It's 32 bits, it could overflow in theory.

Reply:
Yes, but you need `(1 << 32) - 1` operationId collisions, which are difficult to mine and I don't see anyone overflowing this. Even doing it intentionally doesn't seem problematic for me. Even with a 1 day full-governance cycle (propose+vote+queue), it'll take ~11 million years.

Reply:
Note that this is in AccessManager so it doesn't need to go through governance cycle. If it was a 10 minute delay (the smallest that I think might be reasonable), you could schedule once every 10 minutes, how long would it take to overflow?

Reply:
True, this is not related to governance. In the case of an execution every 10 minutes it'll take ~81,000 years (`2**32*10/60/24/365`), it'll be even ~8,000 years if it's every minute or ~1634 years if done every block of 12 seconds (`2**32*12/60/60/24/365`)

Reply:
Adding a comment to explain use of unchecked following our guidelines:

```suggestion
        unchecked {
            // Overflowing the nonce would take over 1000 years
            nonce = _schedules[operationId].nonce + 1;
        }
```